### PR TITLE
test(mme): Add MME_APP unit tests for path switch request

### PR DESCRIPTION
**Pull request by ssanadhya**
Signed-off-by: Shruti Sanadhya <ssanadhya@fb.com>

## Summary

This change adds success and failure test case for MME_APP handlers for Path Switch Request. 

## Test Plan

make test_oai

### CODE REVIEW

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp [hunk 1]
```diff
@@ -2886,5 +2913,222 @@ TEST_F(MmeAppProcedureTest, TestTauRejDueToInvalidTac) {
...
+  // bearer response is received
+  EXPECT_CALL(
+      *s1ap_handler,
+      s1ap_handle_path_switch_req_ack(check_params_in_path_switch_req_ack(
+          new_enb_ue_s1ap_id, new_sctp_assoc_id)))
+      .Times(1);
+
+  // Constructing and sending Modify Bearer Response to mme_app
+  // mimicing SPGW, with same parameters as last one
+  send_modify_bearer_resp(b_modify, b_rm);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp:3237]: shall we check the state again here?

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp [hunk 2]
```diff
@@ -2886,5 +2913,222 @@ TEST_F(MmeAppProcedureTest, TestTauRejDueToInvalidTac) {
...
+  EXPECT_CALL(
+      *s1ap_handler, s1ap_handle_path_switch_req_failure(
+                         check_params_in_path_switch_req_failure(
+                             new_enb_ue_s1ap_id, new_sctp_assoc_id)))
+      .Times(1);
+
+  // Constructing and sending Modify Bearer Response to mme_app
+  // mimicing SPGW, with empty modify bearer list to trigger failure
+  b_modify = {};
+  send_modify_bearer_resp(b_modify, b_rm);
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp:3356]: shall we check the expected state after MBR response?

**Review by ulaskozat — APPROVED** [review]
LGTM, left couple of comments.

**Comment by ulaskozat** [comment]
merge conflict?

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp (hunk 1 above)
Comment (ssanadhya) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp:3237]: Check added

File: lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp (hunk 2 above)
Comment (ssanadhya) [reply lte/gateway/c/core/oai/test/mme_app_task/test_mme_procedures.cpp:3356]: Check added

**Comment by ssanadhya** [comment]
> merge conflict?

Rebased to master and resolved conflict

**Pull request merged**